airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -1095,6 +1099,11 @@ def add_to_outputs(
             else:
                 # incremental is handled in the SCD SQL already
                 sql = self.add_incremental_clause(sql)
+                if do_deletions:
+                    # TODO figure out how to get SCD table name correctly
+                    config[
+                        "post_hook"
+                    ] = f"""["delete from {{{{ this }}}} where _airbyte_ab_id in ( select _airbyte_ab_id from {{{{ {scd_table} }}}} where _airbyte_active_row = 0 and cast({self.name_transformer.normalize_column_name(self.airbyte_normalized_at)} as {{{{ type_timestamp_with_timezone() }}}}) >= (select max(cast({self.name_transformer.normalize_column_name(self.airbyte_normalized_at)} as {{{{ type_timestamp_with_timezone() }}}})) from {{{{ {scd_table} }}}}))"]"""
```

Comment:
Is this a post-hook on the final table to delete from scd table that is built before the final table?

that does not seem right or i am not understanding this correctly?

Reply:
`{{ this }}` is the final table in this context. E.g. from dbt.log:
```sql
with partitioned_data as (
    select row_number() over (
        partition by _airbyte_unique_key
        order by _airbyte_active_row desc, _airbyte_ab_id
    ) as _airbyte_row_num,
    "postgres".test_normalization."dedup_exchange_rate_scd".*
    from "postgres".test_normalization."dedup_exchange_rate_scd"
    where 1=1
    
and coalesce(
    cast(_airbyte_emitted_at as 
    timestamp with time zone
) >= (select max(cast(_airbyte_emitted_at as 
    timestamp with time zone
)) from "postgres".test_normalization."dedup_exchange_rate_scd"),
    
    
    true)

)
delete from "postgres".test_normalization."dedup_exchange_rate" where _airbyte_unique_key in (
    select _airbyte_unique_key from partitioned_data
    where _airbyte_active_row = 0 and _airbyte_row_num = 1
)
```

(i.e. it's deleting from `dedup_exchange_rate` rather than `dedup_exchange_rate_scd`)